### PR TITLE
[CWS] Fix activity tree process cookie bug

### PR DESCRIPTION
**Pull request by spikat**
### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by hestonhoffman — APPROVED** [review]

File: pkg/security/security_profile/activity_tree/activity_tree.go [hunk 1]
```diff
@@ -92,8 +97,8 @@ type ActivityTree struct {
 	validator    ActivityTreeOwner
 	pathsReducer *PathsReducer
 
-	CookieToProcessNode map[uint32]*ProcessNode `json:"-"`
-	ProcessNodes        []*ProcessNode          `json:"-"`
+	CookieToProcessNode map[cookieSelector]*ProcessNode `json:"-"`
```
Comment (safchain) [inline pkg/security/security_profile/activity_tree/activity_tree.go:100]: are the json tags required ?

File: pkg/security/security_profile/activity_tree/activity_tree.go [hunk 2]
```diff
@@ -359,18 +364,23 @@ func (at *ActivityTree) CreateProcessNode(entry *model.ProcessCacheEntry, branch
...
 		if found {
 			return node, false, nil
 		}
 	}
 
 	defer func() {
 		// if a node was found, and if the entry has a valid cookie, insert a cookie shortcut
-		if entry.Cookie > 0 && node != nil {
-			at.CookieToProcessNode[entry.Cookie] = node
+		if cs.cookie != 0 && node != nil {
```
Comment (Gui774ume) [inline pkg/security/security_profile/activity_tree/activity_tree.go:382]: I think you need to check both the exec time and the cookie again here :( => should we create a method for `cookieSelector` ?

File: pkg/security/security_profile/activity_tree/activity_tree.go [hunk 3]
```diff
@@ -359,18 +364,23 @@ func (at *ActivityTree) CreateProcessNode(entry *model.ProcessCacheEntry, branch
 	}
 
 	// look for a ProcessActivityNode by process cookie
-	if entry.Cookie > 0 {
+	cs := cookieSelector{}
+	if entry.ExecTime.UnixNano() != 0 && entry.Cookie != 0 {
```
Comment (Gui774ume) [inline pkg/security/security_profile/activity_tree/activity_tree.go:368]: I think we should rather use `!entry.ExecTime.IsZero()` if possible 🤔

Comment (spikat) [reply pkg/security/security_profile/activity_tree/activity_tree.go]: Suggestion applied, thanks

File: pkg/security/security_profile/activity_tree/activity_tree.go (hunk 2 above)
Comment (spikat) [reply pkg/security/security_profile/activity_tree/activity_tree.go]: Suggestion applied, thanks

File: pkg/security/security_profile/activity_tree/activity_tree.go (hunk 1 above)
Comment (spikat) [reply pkg/security/security_profile/activity_tree/activity_tree.go]: Suggestion applied, thanks

File: pkg/security/security_profile/activity_tree/activity_tree.go [hunk 4]
```diff
@@ -347,6 +359,13 @@ func GetNextAncestorBinaryOrArgv0(entry *model.ProcessContext) *model.ProcessCac
 	return nil
 }
 
+func eventHaveValidCookie(entry *model.ProcessCacheEntry) bool {
+	if !entry.ExecTime.IsZero() && entry.Cookie != 0 {
```
Comment (safchain) [inline pkg/security/security_profile/activity_tree/activity_tree.go:363]: nit: you could just do return `!entry.ExecTime.IsZero() && entry.Cookie != 0` ?

File: pkg/security/security_profile/activity_tree/activity_tree.go [hunk 5]
```diff
@@ -81,6 +81,18 @@ type ActivityTreeOwner interface {
 	NewProcessNodeCallback(p *ProcessNode)
 }
 
+type cookieSelector struct {
+	execTime int64
+	cookie   uint32
+}
+
+func (cs *cookieSelector) isSet() bool {
+	if cs.execTime != 0 && cs.cookie != 0 {
```
Comment (safchain) [inline pkg/security/security_profile/activity_tree/activity_tree.go:90]: you could return `cs.execTime != 0 && cs.cookie != 0`

**Review by safchain — APPROVED** [review]

File: pkg/security/security_profile/activity_tree/activity_tree.go (hunk 4 above)
Comment (spikat) [reply pkg/security/security_profile/activity_tree/activity_tree.go]: Suggestion applied, thanks

File: pkg/security/security_profile/activity_tree/activity_tree.go (hunk 5 above)
Comment (spikat) [reply pkg/security/security_profile/activity_tree/activity_tree.go]: Suggestion applied, thanks

**Pull request merged**